src/wasm-emscripten.h

```diff
@@ -71,6 +75,10 @@ class EmscriptenGlueGenerator {
   // so far.
   std::unordered_set<std::string> sigs;
 
+  // Created by fixEmAsm and used by generateEmscriptenMetadata.
```

Comment:
This comment feels like it will get out of date, because it's saying where in the code these fields are used. Prefer comments that say "why", not "what". In this case, what do these fields mean? What are they used *for*?

My understanding of the code would comment something like: "Stores metadata from processing EM_ASM calls. We store it to decouple the timing of changing the IR from emitting the metadata."
If your understanding of the why is better (it probably is), edit that like crazy.